server/src/main/java/org/elasticsearch/index/mapper/NumberFieldMapper.java

```diff
@@ -734,6 +842,7 @@ public final NumericType numericType() {
         public final TypeParser parser() {
             return parser;
         }
+        public abstract BiFunction<Script, ScriptCompiler, MapperScript> compiler(String fieldName);
```

Comment:
hopefully this can become mapperScript(String fieldName, Script script, ScriptCompiler compiler) ?